Xamarin.Forms.Platform.iOS/Renderers/NavigationRenderer.cs

```diff
@@ -319,6 +319,10 @@ protected virtual void OnElementChanged(VisualElementChangedEventArgs e)
 
 		protected virtual async Task<bool> OnPushAsync(Page page, bool animated)
 		{
+			if(page is MasterDetailPage)
+				System.Diagnostics.Debug.WriteLine($"Pushing a {nameof(MasterDetailPage)} onto a {nameof(NavigationPage)} is not a supported UI pattern on iOS. " +
```

Comment:
You're right. I should have been more careful. I'm not sure if `Trace` is the right thing to use here. I can't seem to find an example where they write to the console for production-use.